lib/modern_searchlogic/defined_scope_set.rb

```diff
@@ -0,0 +1,29 @@
+class DefinedScopeSet
```

Comment:
this class seems like overkill for this change, i'd just call `to_sym` inside of `valid_searchlogic_scope?` in `ModernSearchLogic::ColumnConditions::ClassMethods`

if we are gonna go this route I'd prefer using something like `HashWithIndifferentAccess` which we don't have to roll our own version of

Reply:
That totally works, it was the third time we were calling `to_sym` on `name` so it seemed like it might be a good time to have that be handled automatically but I don't have a strong opinion on it.